File: Java/Sample_Source/DocumentConversion/CreateDocFromXPS/src/com/datalogics/PDFL/Samples/CreateDocFromXPS.java
package com.datalogics.PDFL.Samples;

import java.util.EnumSet;

import com.datalogics.PDFL.Document;
import com.datalogics.PDFL.Library;
import com.datalogics.PDFL.SaveFlags;
import com.datalogics.PDFL.XPSConvertParams;

/*
* This sample demonstrates converting an XPS file into a PDF document.
 * 
 * XML Paper Specification (XPS) is a standard document format that Microsoft created in 2006
 * as an alternative to the PDF format.  
 *
 * Copyright (c) 2007-2010, Datalogics, Inc. All rights reserved.
 *
 * The information and code in this sample is for the exclusive use of Datalogics
 * customers and evaluation users only.  Datalogics permits you to use, modify and
 * distribute this file in accordance with the terms of your license agreement.
 * Sample code is for demonstrative purposes only and is not intended for production use.
 *
 */

public class CreateDocFromXPS {

	/**
	 * @param args
	 */
	public static void main(String[] args) throws Throwable {
        System.out.println("CreateDocFromXPS sample:");

    	Library lib = new Library();

		try {
            // First, create an XPSConvertParams to specify conversion parameters 
            // for creating the document.
            XPSConvertParams xpsparams = new XPSConvertParams();
            
            // DLE requires a .joboptions file to specify settings for XPS conversion. 
            // A default .joboptions file is provided in the Resources directory of 
            // the DLE distribution.  This file is used by default, but a custom file
            // can be used instead by setting the pathToSettingsFile property.
            System.out.println("Using settings file located at: " + xpsparams.getPathToSettingsFile());
            
            // Create the document.
            System.out.println("Creating a document from an XPS file...");
	        Document doc = new Document("../../Resources/Sample_Input/brownfox.xps", xpsparams);
	        
            // Save the document.
            System.out.println("Saving the document...");
            doc.save(EnumSet.of(SaveFlags.FULL), "CreateDocFromXPS-out.pdf");                
		}
		finally {
			lib.delete();
		}
	}

}

File: DotNET/Sample_Source/DocumentConversion/CreateDocFromXPS/CreateDocFromXPS.cs
using System;
using System.Collections.Generic;
using System.Text;
using Datalogics.PDFL;

/*
 * This sample demonstrates converting an XPS file into a PDF document.
 * 
 * XML Paper Specification (XPS) is a standard document format that Microsoft created in 2006
 * as an alternative to the PDF format.  
 *
 * Copyright (c) 2007-2020, Datalogics, Inc. All rights reserved.
 *
 * For complete copyright information, refer to:
 * http://dev.datalogics.com/adobe-pdf-library/license-for-downloaded-pdf-samples/
 *
 */

namespace CreateDocFromXPS
{
    class CreateDocFromXPS
    {
        static void Main(string[] args)
        {
            Console.WriteLine("CreateDocFromXPS sample:");


            using(Library lib = new Library())
            {

                String sInput = Library.ResourceDirectory + "Sample_Input/brownfox.xps";
                String sOutput = "CreateDocFromXPS-out.pdf";

                if (args.Length > 0)
                    sInput = args[0];

                if (args.Length > 1)
                    sOutput = args[1];

                Console.WriteLine("Input file: " + sInput + ", writing to " + sOutput);

                // First, create an XPSConvertParams to specify conversion parameters 
                // for creating the document.
                XPSConvertParams xpsparams = new XPSConvertParams();
                
                // PDFL requires a .joboptions file to specify settings for XPS conversion.
                // A default .joboptions file is provided in the Resources directory of 
                // the PDFL distribution.  This file is used by default, but a custom file
                // can be used instead by setting the pathToSettingsFile property.
                Console.WriteLine("Using settings file located at: " + xpsparams.PathToSettingsFile);
                
                // Create the document.
                Console.WriteLine("Creating a document from an XPS file...");
	            Document doc = new Document(sInput, xpsparams);
    	        
                // Save the document.
                Console.WriteLine("Saving the document...");
                doc.Save(SaveFlags.Full, sOutput);
    		}
        }
    }
}
